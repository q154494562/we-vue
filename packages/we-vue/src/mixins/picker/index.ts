// Mixins
import Toggleable, { factory as ToggleableFactory } from '../toggleable'

// Utils
import mixins, { ExtractVue } from '@/utils/mixins'

export default mixins<
  ExtractVue<[typeof Toggleable]>
>(
  ToggleableFactory('visible', 'update:visible')
).extend({
  name: 'picker',

  props: {
    confirmText: {
      type: String,
      default: '确定',
    },
    cancelText: {
      type: String,
      default: '取消',
    },
    closeOnClickMask: {
      type: Boolean,
      default: true,
    },
  },

  methods: {
    open (): void {
      this.isActive = true
    },

    close (): void {
      this.isActive = false
    },
  },
})
