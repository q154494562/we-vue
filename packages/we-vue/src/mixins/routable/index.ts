import Vue from 'vue'

export default Vue.extend({
  name: 'routable',

  props: {
    href: String,
    replace: Boolean,
    to: [String, Object],
  },

  methods: {
    routeLink () {
      const { to, href, $router, replace } = this as any
      if (to && $router) {
        $router[replace ? 'replace' : 'push'](to)
      } else if (href) {
        replace ? location.replace(href) : location.href = href
      }
    },
  },
})
