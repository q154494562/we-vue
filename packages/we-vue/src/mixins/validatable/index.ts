import Vue from 'vue'

export default Vue.extend({
  name: 'validatable',

  props: {
    value: {
      required: false,
    },
  },

  data () {
    return {
      lazyValue: this.value,
      valid: false,
    }
  },
})
