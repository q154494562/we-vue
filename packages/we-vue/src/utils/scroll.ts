type ElementOrWindow = HTMLElement | Window

/**
 * 找到最近的触发滚动事件的元素
 * @param {Element} element
 * @param {Element} rootParent
 * @returns {Element | window}
 */
export function getScrollEventTarget (element: HTMLElement, rootParent: ElementOrWindow = window) {
  let node = element
  while (
    node &&
    node.tagName !== 'HTML' &&
    node.tagName !== 'BODY' &&
    node.nodeType === 1 &&
    node !== rootParent
  ) {
    const { overflowY } = window.getComputedStyle(node)
    if (overflowY === 'scroll' || overflowY === 'auto') {
      return node
    }
    node = <HTMLElement>node.parentNode
  }
  return rootParent
}

// 获取滚动高度
export function getScrollTop (element: ElementOrWindow): number {
  return 'scrollTop' in element ? element.scrollTop : element.pageYOffset
}

// 设置滚动高度
export function setScrollTop (element: ElementOrWindow, value: number) {
  'scrollTop' in element ? (element.scrollTop = value) : element.scrollTo(element.scrollX, value)
}

// 获取元素距离顶部高度
export function getElementTop (element: ElementOrWindow) {
  return (
    (element === window ? 0 : (element as HTMLElement).getBoundingClientRect().top) + getScrollTop(window)
  )
}

export function getVisibleHeight (element: ElementOrWindow) {
  return element === window
    ? element.innerHeight
    : (element as HTMLElement).getBoundingClientRect().height
}
