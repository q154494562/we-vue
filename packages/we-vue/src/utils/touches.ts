function getTouch (event: TouchEvent): Touch {
  return event.changedTouches[0] || event.touches[0]
}

export { getTouch }
