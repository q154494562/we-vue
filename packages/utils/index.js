import Vue from 'vue'
import { getTouch } from './touches'

const isServer = Vue.prototype.$isServer

function isDef (value) {
  return value !== undefined && value !== null
}

function isObj (x) {
  const type = typeof x
  return x !== null && (type === 'object' || type === 'function')
}

export {
  isServer,
  getTouch,
  isDef,
  isObj
}
