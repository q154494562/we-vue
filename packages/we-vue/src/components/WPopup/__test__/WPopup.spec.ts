import { mount, Wrapper } from '@vue/test-utils'
import Popup from '../'
import { ExtractVue } from '@/utils/mixins'

describe('popup', () => {
  type Instance = ExtractVue<typeof Popup>
  let mountFunction: (options?: object) => Wrapper<Instance>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(Popup, {
        ...options,
      })
    }
  })

  test('create', () => {
    const wrapper = mount(Popup, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-popup')
    expect(wrapper.classes()).toContain('wv-popup')
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('create a popup that is visible', () => {
    const wrapper = mount(Popup, {
      attachToDocument: true,
      propsData: {
        visible: true,
      },
    })

    expect(wrapper.classes()).toContain('wv-popup')

    expect(wrapper.isVisible()).toBeTruthy()
  })

  test('computed style', () => {
    const wrapper = mountFunction({
      propsData: {
        height: 100,
      },
    })

    expect(wrapper.vm.style).toEqual({
      backgroundColor: wrapper.vm.backgroundColor,
      height: '100px',
    })
  })
})
