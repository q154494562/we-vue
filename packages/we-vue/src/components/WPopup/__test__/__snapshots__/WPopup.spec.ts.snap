// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup create 1`] = `<div class="wv-popup" style="height: auto; display: none;"></div>`;
